src/balancer-strategy-manager/BalancerV2WeightedPoolStrategyManager.sol

```diff
@@ -0,0 +1,233 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import {IERC20} from 'solidity-utils/contracts/oz-common/interfaces/IERC20.sol';
+import {SafeERC20} from 'solidity-utils/contracts/oz-common/SafeERC20.sol';
+import {OwnableWithGuardian} from 'solidity-utils/contracts/access-control/OwnableWithGuardian.sol';
+import {Rescuable} from 'solidity-utils/contracts/utils/Rescuable.sol';
+import {RescuableBase, IRescuableBase} from 'solidity-utils/contracts/utils/RescuableBase.sol';
+
+import {IBalancerVault} from './balancer-v2/IBalancerVault.sol';
+import {WeightedPoolUserData} from './balancer-v2/WeightedPoolUserData.sol';
+import {IBalancerStrategyManager} from './IBalancerStrategyManager.sol';
+
+/**
+ * @title Balancer V2 Weighted Pool Strategy Manager
+ * @author TokenLogic
+ * @notice A contract to manage deposits and withdrawals into a Balancer V2 Weighted Pool.
+ * @dev This contract uses the Balancer Vault to manage liquidity, using exact token amounts for BPT.
+ */
+contract BalancerV2WeightedPoolStrategyManager is
+  IBalancerStrategyManager,
+  OwnableWithGuardian,
+  Rescuable
+{
+  using SafeERC20 for IERC20;
+
+  /// @notice The Balancer Vault contract
+  IBalancerVault public immutable VAULT;
+
+  /// @notice The number of tokens in the pool
+
+  /// @notice The address of the Hypernative service
+  address public immutable HYPERNATIVE;
+
+  /// @dev Mapping of token configurations (token to provider)
+  mapping(address token => address provider) public override tokenProvider;
```

Comment:
What are we overriding?

Reply:
This function defined on interface. and provider is sth like treasury. I'm not sure how can I name this. :)